src/internal/bind-to-namespace.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Bind all methods of a class to the class instance.
+ *
+ * @param moduleClass The class to bind the methods of.
+ * @param that The class instance to bind the methods to.
+ */
+export function bindToNamespace<TClass extends { new (...args: any[]): any }>(
+  moduleClass: TClass,
+  that: InstanceType<TClass>
+): void {
+  // Bind `this` so namespaced is working correctly
+  for (const name of Object.getOwnPropertyNames(moduleClass.prototype) as Array<
```

Comment:
Im pretty sure that you could completely get rid of the class argument in this function by doing the following here:


```suggestion
  for (const name of Object.getOwnPropertyNames(Object.getPrototypeOf(that)) as Array<
```

I'd suggest this as it is quite confusing to me that you have to explicitly have to pass the class reference which you are currently instantiating.

[POC in TS Playground](https://tsplay.dev/WYlxQm) (run and check the console)

Reply:
I tested the changes with the faker project and it worked well.

Reply:
Do we still need the explicit cast?